pkg/operator/manager.go

```diff
@@ -53,16 +49,16 @@ type HealthCheck interface {
 }
 
 // NewManagerOrDie instantiates a controller manager or panics
-func NewManagerOrDie(ctx context.Context, config *rest.Config, opts *options.Options) manager.Manager {
+func NewManagerOrDie(ctx context.Context, config *rest.Config, options *Options) manager.Manager {
 	newManager, err := controllerruntime.NewManager(config, controllerruntime.Options{
 		Logger:                     ignoreDebugEvents(zapr.NewLogger(logging.FromContext(ctx).Desugar())),
-		LeaderElection:             opts.EnableLeaderElection,
+		LeaderElection:             options.EnableLeaderElection,
 		LeaderElectionID:           "karpenter-leader-election",
 		LeaderElectionResourceLock: resourcelock.LeasesResourceLock,
 		Scheme:                     scheme,
-		MetricsBindAddress:         fmt.Sprintf(":%d", opts.MetricsPort),
-		HealthProbeBindAddress:     fmt.Sprintf(":%d", opts.HealthProbePort),
-		BaseContext:                newRunnableContext(config, opts, logging.FromContext(ctx)),
+		MetricsBindAddress:         fmt.Sprintf(":%d", options.MetricsPort),
+		HealthProbeBindAddress:     fmt.Sprintf(":%d", options.HealthProbePort),
+		BaseContext:                func() context.Context { return ctx },
```

Comment:
Base context was introduced to avoid top-level context cancellation from stopping runnable cleanup as shown [here](https://github.com/kubernetes-sigs/controller-runtime/pull/1846). We shouldn't be passing the top-level context in a closure here since this gets back to the issue that controller-runtime was trying to avoid

Reply:
Fixed and added a comment